crates/holochain/src/conductor/conductor.rs

```diff
@@ -1777,9 +1777,70 @@ mod scheduler_impls {
 
 /// Miscellaneous methods
 mod misc_impls {
+    use holochain_zome_types::builder;
+
+    use crate::conductor::api::error::ConductorApiError;
+
     use super::*;
 
     impl Conductor {
+        /// Authorize a zome call signing key for a cell
+        pub async fn authorize_zome_call_signing_key(
+            &self,
+            payload: AuthorizeZomeCallSigningKeyPayload,
+        ) -> ConductorApiResult<()> {
+            let AuthorizeZomeCallSigningKeyPayload {
+                agent_pub_key,
+                cell_id,
+                cap_grant,
+            } = payload;
+
+            if agent_pub_key != *cell_id.agent_pubkey() {
+                return Err(ConductorApiError::IllegalZomeCallSigningKeyAuthorization(
+                    cell_id.clone(),
+                    agent_pub_key.clone(),
+                ));
+            }
```

Comment:
If the agent pub key must match the one in the cell id, then I think there's no point in passing it in as a parameter, might as well just pull the agent pub key from the cell id, right?

Reply:
true that!